Mage.Sets/src/mage/cards/c/CaptainAmericaFirstAvenger.java

```diff
@@ -0,0 +1,215 @@
+    }
+}
+
+enum CaptainAmericaFirstAvengerValue implements DynamicValue {
+    instance;
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        int amount = 0;
+        for (Cost cost : sourceAbility.getCosts()) {
+            if (cost instanceof CaptainAmericaFirstAvengerUnattachCost && !cost.getTargets().isEmpty()) {
+                Permanent equipment = game.getPermanentOrLKIBattlefield(cost.getTargets().getFirstTarget());
+                if (equipment != null) {
+                    amount = equipment.getManaValue();
+                }
+            }
+        }
+        return amount;
+    }
+
+    @Override
+    public DynamicValue copy() {
+        return instance;
+    }
+
+    @Override
+    public String toString() {
+        return "X";
+    }
+
+    @Override
+    public String getMessage() {
+        return "that Equipment's mana value";
+    }
+}
+
+class CaptainAmericaFirstAvengerUnattachCost extends EarlyTargetCost {
+
+    private static final FilterPermanent filter = new FilterEquipmentPermanent("equipment attached to this creature");
+    private static final FilterPermanent subfilter = new FilterControlledPermanent("{this}");
+
+    static {
+        subfilter.add(CaptainAmericaPredicate.instance);
+        filter.add(new AttachedToPredicate(subfilter));
+    }
+
+    public CaptainAmericaFirstAvengerUnattachCost() {
+        super();
+    }
+
+    protected CaptainAmericaFirstAvengerUnattachCost(final CaptainAmericaFirstAvengerUnattachCost cost) {
+        super(cost);
+    }
+
+    @Override
+    public boolean canPay(Ability ability, Ability source, UUID controllerId, Game game) {
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        return permanent != null
+                && !permanent.getAttachments().isEmpty();
+    }
+
+    @Override
+    public boolean pay(Ability ability, Game game, Ability source, UUID controllerId, boolean noMana, Cost costToPay) {
+        Player player = game.getPlayer(source.getControllerId());
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        if (permanent == null) {
+            return paid;
+        }
+        Permanent equipment = game.getPermanentOrLKIBattlefield(getTargets().getFirstTarget());
+        if (equipment == null || !permanent.getAttachments().contains(getTargets().getFirstTarget()) ||
+                !player.chooseUse(Outcome.Benefit, "Unattach " + equipment.getName() + "?", source, game)) {
+            return false;
+        }
+        paid = permanent.removeAttachment(getTargets().getFirstTarget(), source, game);
+
+        return paid;
+    }
+
+    @Override
+    public CaptainAmericaFirstAvengerUnattachCost copy() {
+        return new CaptainAmericaFirstAvengerUnattachCost(this);
+    }
+
+    @Override
+    public void chooseTarget(Game game, Ability source, Player controller) {
+        Target chosenEquipment = new TargetPermanent(1, 1, filter, true);
```

Comment:
does this need to clear targets first?

is it possible to abstract any part of this method into the general logic to reduce what each `EarlyTargetCost` needs to implement?

Reply:
No, `Cost`s are initialized with null `targets` and then the first time an instance of a `Cost` has the targets either `Get` or `Set` they initialize it with an empty `Targets`, so there won't be anything to clear; and it doesn't `chooseTarget` in `handleChooseCostTargets` unless `cost.getTargets().isEmpty()`

Reply:
Just checking because this method is outside the normal target process.

If the cost is to be paid a second time on a subsequent activation, is it in a fresh state with empty targets?

Reply:
It is, yes.